python/paddle/fluid/dygraph/jit.py

```diff
@@ -210,7 +210,16 @@ def decorated(python_func):
 
     # for usage: `declarative(foo, ...)`
     if function is not None:
-        return decorated(function)
+        if isinstance(function, Layer):
+            if isinstance(function.forward, StaticLayer):
```

Comment:
Please check whether ``function`` has ``forward``, if not, I think we should throw our exception to indicate users?

Reply:
Sorry, I was wrong, ``function`` is an instance of ``Layer`` so it must have the ``forward``. I will approve this PR

Reply:
It inherit `Layer` and hold `forward` function.